apps/asap-server/src/handlers/webhooks/webhook-run-migrations.ts

```diff
@@ -0,0 +1,257 @@
+  const filterUnexecutedMigrations = filterUnexecutedMigrationsFactory(client);
+  const saveExecutedMigrations = saveExecutedMigrationsFactory(client);
+  const getMigrationsFromPaths = getMigrationsFromPathsFactory(
+    importModule,
+    logger,
+  );
+
+  const migrationPaths = await getMigrationPaths();
+  const unexecutedMigrationPaths = await filterUnexecutedMigrations(
+    migrationPaths,
+  );
+
+  logger.debug(
+    { migrations: unexecutedMigrationPaths },
+    'Outstanding migrations',
+  );
+
+  const migrations = await getMigrationsFromPaths(unexecutedMigrationPaths);
+
+  const executedMigrations: string[] = [];
+
+  for (const migration of migrations) {
+    logger.debug(`Executing migration '${migration.getPath()}`);
+    try {
+      await migration.up();
+      executedMigrations.push(migration.getPath());
+
+      logger.info(`Executed migration '${migration.getPath()}`);
+    } catch (error) {
+      logger.error(
+        error,
+        `Error executing the migration ${migration.getPath()}`,
+      );
+      break;
+    }
+  }
+
+  logger.debug('Finished executing migrations, saving progress...');
+
+  await saveExecutedMigrations(executedMigrations);
+
+  logger.info(`Executed and saved ${executedMigrations.length} migrations`);
+};
+
+export const rollbackFactory = (
+  logger: Logger,
+  client: Squidex<RestMigration>,
+  readDir: typeof fsPromise.readdir,
+  importModule: ImportModuleFromPath,
+): Handler => async () => {
+  // const getMigrationPaths = getMigrationPathsFromDirectoryFactory(readDir);
+  const getLatestMigrationPathFromDb = getLatestMigrationPathFromDbFactory(
+    client,
+  );
+  const getMigrationsFromPaths = getMigrationsFromPathsFactory(
+    importModule,
+    logger,
+  );
+  const removeExecutedMigration = removeExecutedMigrationFactory(client);
+
+  const migrationPath = await getLatestMigrationPathFromDb();
+
+  logger.debug({ migrations: migrationPath }, 'Latest migrations');
+
+  if (migrationPath !== null) {
+    let migration: Migration;
+    try {
+      [migration] = await getMigrationsFromPaths([migrationPath]);
+    } catch (error) {
+      logger.error(
+        error,
+        `Could not load the migration from file ${migrationPath}`,
+      );
+
+      return;
+    }
+
+    logger.debug(`Executing rollback of migration '${migration.getPath()}`);
+
+    try {
+      await migration.down();
+    } catch (error) {
+      logger.error(
+        error,
+        `Error executing the rollback of migration ${migration.getPath()}`,
+      );
+      return;
+    }
+
+    logger.debug('Finished executing rollback, saving progress...');
+
+    await removeExecutedMigration(migrationPath);
+
+    logger.info(`Rolled back and removed migration '${migration.getPath()}'`);
+  }
+};
+
+const getMigrationPathsFromDirectoryFactory = (
+  readDir: typeof fsPromise.readdir,
+) => async () => (await readDir(migrationDir)).sort();
```

Comment:
How are the migrations getting into the filesystem of the lambda? We sure these are not getting excluded by tree-shaking?

Reply:
nah i tested it, somehow it resolves dynamic imports